Support 5.6 kernels as on Fedora 31

Now assurio-snap build is failing on Fedora 31 with the kernel 5.6.7-200.

```
make -C /lib/modules/5.6.7-200.fc31.x86_64/build M=/home/assurio/assurio-snap/src modules
make[2]: Entering directory '/usr/src/kernels/5.6.7-200.fc31.x86_64'
  CC [M]  /home/assurio/assurio-snap/src/assurio-snap.o
/home/assurio/assurio-snap/src/assurio-snap.c: In function ‘agent_init’:
/home/assurio/assurio-snap/src/assurio-snap.c:5166:51: error: passing argument 4 of ‘proc_create’ from incompatible pointer type [-Werror=incompatible-pointer-types]
 5166 |  info_proc = proc_create(INFO_PROC_FILE, 0, NULL, &assurio_snap_proc_fops);
      |                                                   ^~~~~~~~~~~~~~~~~~~~~~~
      |                                                   |
      |                                                   const struct file_operations *
In file included from /home/assurio/assurio-snap/src/includes.h:17,
                 from /home/assurio/assurio-snap/src/assurio-snap.c:8:
./include/linux/proc_fs.h:64:24: note: expected ‘const struct proc_ops *’ but argument is of type ‘const struct file_operations *’
   64 | struct proc_dir_entry *proc_create(const char *name, umode_t mode, struct proc_dir_entry *parent, const struct proc_ops *proc_ops);
      |                        ^~~~~~~~~~~
cc1: some warnings being treated as errors
make[3]: *** [scripts/Makefile.build:268: /home/assurio/assurio-snap/src/assurio-snap.o] Error 1
make[2]: *** [Makefile:1683: /home/assurio/assurio-snap/src] Error 2
make[2]: Leaving directory '/usr/src/kernels/5.6.7-200.fc31.x86_64'
make[1]: *** [Makefile:14: default] Error 2
make[1]: Leaving directory '/home/assurio/assurio-snap/src'
make: *** [Makefile:24: driver] Error 2
[assurio@fedora31-build assurio-snap]$ grep -rnw file_operations
src/assurio-snap.c:585:static inline struct proc_dir_entry *proc_create(const char *name, mode_t mode, struct proc_dir_entry *parent, const struct file_operations *proc_fops){
src/assurio-snap.c:901:static const struct file_operations snap_control_fops = {
src/assurio-snap.c:922:static const struct file_operations assurio_snap_proc_fops = {

```

--- a/src/configure-tests/feature-tests/proc_create_fn_file_operations.c
+++ b/src/configure-tests/feature-tests/proc_create_fn_file_operations.c
@@ -5,9 +5,12 @@
  * Additional contributions by Elastio Software, Inc are Copyright (C) 2020 Elastio Software Inc.
  */
 
+// 2.6.25 <= kernel_version < 5.6
+
 #include "includes.h"
 
 static inline void dummy(void){
-	struct proc_dir_entry *ent = proc_create("file", 0, NULL, NULL);
+ 	static const struct file_operations file_ops;
+	struct proc_dir_entry *ent = proc_create("file", 0, NULL, &file_ops);
 	(void)ent;
 }

--- a/src/configure-tests/feature-tests/proc_create_fn_proc_ops.c
+++ b/src/configure-tests/feature-tests/proc_create_fn_proc_ops.c
@@ -1,0 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/*
+ * Copyright (C) 2020 Elastio Software Inc.
+ */
+
+// 5.6 <= kernel_version
+
+#include "includes.h"
+
+static inline void dummy(void){
+ 	static const struct proc_ops file_ops;
+	struct proc_dir_entry *ent = proc_create("file", 0, NULL, &file_ops);
+	(void)ent;
+}

--- a/src/elastio-snap.c
+++ b/src/elastio-snap.c
@@ -580,7 +580,7 @@ static inline void elastio_snap_inode_unlock(struct inode *inode){
 	#define elastio_snap_inode_unlock inode_unlock
 #endif
 
-#ifndef HAVE_PROC_CREATE
+#if !defined HAVE_PROC_CREATE_FN_FILE_OPERATIONS && !defined HAVE_PROC_CREATE_FN_PROC_OPS
 //#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,25)
 static inline struct proc_dir_entry *proc_create(const char *name, mode_t mode, struct proc_dir_entry *parent, const struct file_operations *proc_fops){
 	struct proc_dir_entry *ent;
@@ -919,6 +919,14 @@ static const struct seq_operations elastio_snap_seq_proc_ops = {
 	.show = elastio_snap_proc_show,
 };
 
+#ifdef HAVE_PROC_CREATE_FN_PROC_OPS
+static const struct proc_ops elastio_snap_proc_fops = {
+	.proc_open = elastio_snap_proc_open,
+	.proc_read = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = elastio_snap_proc_release,
+};
+#else
 static const struct file_operations elastio_snap_proc_fops = {
 	.owner = THIS_MODULE,
 	.open = elastio_snap_proc_open,
@@ -926,6 +934,7 @@ static const struct file_operations elastio_snap_proc_fops = {
 	.llseek = seq_lseek,
 	.release = elastio_snap_proc_release,
 };
+#endif
 
 static int major;
 static struct mutex ioctl_mutex;

